jetty-ee10/jetty-ee10-tests/jetty-ee10-test-client-transports/src/test/java/org/eclipse/jetty/ee10/test/client/transport/HttpClientContinueTest.java

```diff
@@ -113,17 +114,83 @@ protected void service(HttpServletRequest request, HttpServletResponse response)
+    public void testExpect100ContinueWithMultipleContentsRespond100ContinueBlocking(Transport transport) throws Exception
+    {
+        testExpect100ContinueRespond100ContinueBlocking(transport, "data1".getBytes(StandardCharsets.UTF_8), "data2".getBytes(StandardCharsets.UTF_8), "data3".getBytes(StandardCharsets.UTF_8));
+    }
+
+    private void testExpect100ContinueRespond100ContinueBlocking(Transport transport, byte[]... contents) throws Exception
     {
         start(transport, new HttpServlet()
         {
             @Override
             protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException
             {
                 // Send 100-Continue and copy the content back
+                IO.copy(request.getInputStream(), response.getOutputStream());
+            }
+        });
+
+        ContentResponse response;
+        try (AsyncRequestContent content = new AsyncRequestContent())
+        {
+            new Thread(() ->
+            {
+                for (byte[] b : contents)
+                {
+                    try
+                    {
+                        // Sleep here to ensure that sometimes the reader will block/pause even after sending 100.
+                        Thread.sleep(100);
+                        Callback.Completable callback = new Callback.Completable();
+                        content.write(b == contents[contents.length - 1], ByteBuffer.wrap(b), callback);
+                        callback.get();
+                    }
+                    catch (Throwable t)
+                    {
+                        t.printStackTrace();
+                    }
+                }
+            }).start();
+            response = client.newRequest(newURI(transport))
+                .headers(headers -> headers.put(HttpHeader.EXPECT, HttpHeaderValue.CONTINUE))
+                .body(content)
+                .timeout(5, TimeUnit.SECONDS)
+                .send();
+
+        }
+
+        assertNotNull(response);
+        assertEquals(200, response.getStatus());
+
+        int index = 0;
+        byte[] responseContent = response.getContent();
+        for (byte[] content : contents)
+        {
+            for (byte b : content)
+            {
+                assertEquals(b, responseContent[index++]);
+            }
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("transportsNoFCGI")
+    public void testExpect100ContinueWithChunkedContentRespond100Continue(Transport transport) throws Exception
+    {
+        start(transport, new HttpServlet()
+        {
+            @Override
+            protected void service(HttpServletRequest request, HttpServletResponse response) throws IOException
+            {
+                // read/write first byte to send 100-Continue
                 ServletInputStream input = request.getInputStream();
-                // Make sure we chunk the response too
+                response.getOutputStream().write(input.read());
```

Comment:
I don't like that now we have to read() in order to send the `100` interim response.